tests/utils/test_cli_util.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import unittest
+from argparse import Namespace
+from datetime import datetime
+
+from airflow.utils import cli
+
+
+class CliUtilTest(unittest.TestCase):
+
+    def test_metrics_build(self):
+        func_name = 'test'
+        exec_date = datetime.utcnow()
+        ns = Namespace(dag_id='foo', task_id='bar',
+                       subcommand='test', execution_date=exec_date)
+        metrics = cli._build_metrics(func_name, ns)
+
+        expected = {'user': os.environ.get('USER'),
+                    'sub_command': 'test',
+                    'dag_id': 'foo',
+                    'task_id': 'bar',
+                    'execution_date': exec_date}
+        for k, v in expected.items():
+            self.assertEquals(v, metrics.get(k))
+
+        self.assertTrue(metrics.get('start_datetime'))
```

Comment:
`metrics.get('start_datetime') <= datetime.now()` is a tiny bit better

Reply:
Will do.